### ISSUE TITLE
Pages & Performance Refactor

### ISSUE THREAD

**Issue by azdak** [bbertucc/equalify#494]
Currently most actions and functionality in the app is based on _Properties_, which contain actual individual urls that can only be added by URL **OR** sitemap. This is both confusing to users and makes basic functionality like scanning across subdomains or tracking the status of homepages across several URLs difficult or impossible. This update will make _Pages_ the fundamental entity in Equalify, and make _Properties_ tag-like containers of pages that can be added or removed at will. It will also include performance enhancements to improve app responsiveness.

## Steps to Close this Issue
- [ ] Refactor scan polling system to remove Property dependencies and improve performance 
   - [X] Refactor/Rewrite 
   - [X] Switch to new polling system for ingesting scans from scan.equalify.app 
   - [ ] Test new polling system  [8 Hours]
- [x] New "Pages" screen
   - [X] New getPages API endpoint
   - [X] Pages listing table with serverside pagination
   - [X] New sendPagesToScan API endpoint
   - [X] Enable sending pages to scan from Pages table
   - [x] Pages table styling [4 Hours]
- [x] "Single Page Detail" screen [6 Hours]
   - [x] Scan history table 
- [x] New "Add Pages" screen
   - [X] Add new Add Pages API endpoint
   - [X] Add pages input form, accepting multiple urls or sitemaps
   - [x] Connect input form to API
   - [x] Fix Styling
   - [x] CSV Support [4 Hours]
- [x] "Properties" refactor [8 Hours]
   - [x] Enable adding to and removing pages/sitemaps from Properties
   - [x] List pages on Property Detail screen
- [ ] Remove Property dependencies from any other existing API calls [See note]
   - [x] Error in get/filters endpoint
   - [x] Handle setting properties to null 
   - [x] remove URL and discovery fields from new/edit property
   - [x] add ability to delete URLs
- [ ] Initial Accessibility Testing for added pages and forms [See note]
- [ ] Add "failed" state to scans (May need further discussion)
- [ ] @bbertucc closes issue.

**Comment by bbertucc** [comment]
Thanks. Can you put down a budget and timeline?

**Comment by bbertucc** [comment]
@azdak maybe we need to split this out into smaller tasks that can be estimated? Need to get to a place where we can have an estimate for this work.

**Comment by azdak** [comment]
Update 12/2: I've added time estimates for the outstanding items in the initial buildout, around 30 hours in total. I haven't budgeted for two items, "Remove Property dependencies from any other existing API calls" which will just involve debugging any issues that arise elsewhere in the app with the new Pages-based system, and "Initial Accessibility Testing for added pages and forms" which will likely be a separate issue once the core functionality is complete.

For November, the currently checked/completed items took 22 hours.

**Comment by bbertucc** [comment]
As per our call @azdak is budgeting this.

**Comment by bbertucc** [comment]
Thanks @azdak. As we discussed this is approved.

**Comment by bbertucc** [comment]
Update from today's contributor call: @azdak is working his way through our punch list. Aiming for team to test by December 16.

**Comment by bbertucc** [comment]
Update 12/16: 
- CSV support is done. 
- Browsing isn't crashing anymore
- Added an endpoint for fetching single properties.
- Pushing up changes to properties
- Still needs to do accessibility updates

**Comment by azdak** [comment]
For these two items, estimating 4 hours total:
- [ ] remove URL and discovery fields from new/edit property
- [ ] add ability to delete URLs

**Comment by bbertucc** [comment]
@azdak approved

**Comment by bbertucc** [comment]
I scheduled 12/31 to do functional testing on this.

**Comment by bbertucc** [comment]
@azdak paid you 33 hrs for completed work. I also paid you an additional 30 hour retainer should be billed against.

**Comment by bbertucc** [comment]
Closing this for now because its not part of the current push
